libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -65,6 +70,70 @@ class ImageSource(TypedDict):
     width: int
 
 
+def create_asset_file(
+    dataset: str,
+    config: str,
+    split: str,
+    row_idx: int,
+    column: str,
+    filename: str,
+    assets_base_url: str,
+    assets_directory: StrPath,
+    fn: partial,  # type: ignore
+    overwrite: bool = True,
+    # TODO: Once assets and cached-assets are migrated to S3, this parameter is no more needed
+    use_s3_storage: bool = False,
+    s3_client: Optional[S3Client] = None,
+    # TODO: Once assets and cached-assets are migrated to S3, the following parameters dont need to be optional
+    s3_bucket: Optional[str] = None,
+    s3_folder_name: Optional[str] = None,
+) -> str:
+    dir_path, url_dir_path = get_asset_dir_name(
+        dataset=dataset,
+        config=config,
+        split=split,
+        row_idx=row_idx,
+        column=column,
+        assets_directory=assets_directory,
+    )
+
+    if not use_s3_storage:
+        file_path = dir_path / filename
+        makedirs(dir_path, ASSET_DIR_MODE, exist_ok=True)
+    else:
+        payload = (
+            str(uuid4()),
+            dataset,
+            config,
+            split,
+            row_idx,
+            column,
+        )
+        prefix = sha1(json.dumps(payload, sort_keys=True).encode(), usedforsecurity=False).hexdigest()[:8]
+        file_path = Path(assets_directory).resolve() / f"{prefix}-{filename}"
+
+    if overwrite or not file_path.exists():
+        fn(file_path=file_path)
+
+    if use_s3_storage and s3_client is not None and s3_bucket is not None:
+        object_key = f"{s3_folder_name}/{url_dir_path}/{filename}"
+        create_object = overwrite or not s3_client.exists_in_bucket(s3_bucket, object_key)
```

Comment:
we are checking if the file exists, but maybe it's an outdated version? How could we ensure we always have the correct version file?

Reply:
I haven't realized this issue, Don't we have currently the same problem? I think we could probably add the dataset version to the filename wdyt?

Reply:
No need to fix this is this PR imo. We always do overwrite=True anyway

Reply:
I think I can create another issue to track this behavior.

> We always do overwrite=True
Notice that I changed to overwrite=False in /rows and /search when working with S3 storage. Is that ok? I mean, I think it could help reduce latency when uploading files. Also, I don't think it will make sense to take advantage of Cloudfront when having overwrite=True right?

Reply:
ok